### PR TITLE
Rework bicep scripts. Add in managed identity.

### PR DESCRIPTION
**Pull request by mukundansundar**
## Purpose
Rework AZD bicep scripts to follow template from other samples. 

Add managed identity for service bus. 

Add service bus provisioning as extra script. 

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Review by greenie-msft — APPROVED** [review]

File: azure.yaml [hunk 1]
```diff
@@ -9,7 +9,9 @@ services:
     dist: build
     language: py
     host: containerapp
+    module: app/checkout
```
Comment (paulyuk) [inline azure.yaml]: i'd prefer we do something more generic like `publisher` and `subscriber` in the names throughout.  if we hard code in a "domain object" name like orders then this thing will be so hard to reuse.

**Review by wbreza — COMMENTED** [review]
There have been a good amount of changes to the `containerapp` [bicep core modules](https://github.com/Azure/azure-dev/tree/main/templates/common/infra/bicep/core) from Azd over the last release (released today).  I'd recommend you take a look at those and consider folding them back in here. A [migration guide](https://github.com/Azure/azure-dev/pull/2027) is available in the PR notes.

There is also a pending PR (https://github.com/Azure/azure-dev/pull/2157) that enables dapr configuration as well.

File: azure.yaml (hunk 1 above)
Comment (paulyuk) [reply azure.yaml]: Specifically I think it's ok to leave the code files named orders and checkout.  But the bicep module name should be publisher (the noingress conatiner) and orders (ingress one) in the app folder.  Those in turn set up the container app with and without ingress, and can send orders and checkout as container names as strings.

File: infra/app/app-env.bicep [hunk 2]
```diff
@@ -0,0 +1,58 @@
+param containerAppsEnvName string
+param containerRegistryName string
+param location string
+param logAnalyticsWorkspaceName string
+param serviceBusName string
+param applicationInsightsName string = ''
+param daprEnabled bool = false
+param manegeIdentityClientId string
```
Comment (paulyuk) [inline infra/app/app-env.bicep]:
```suggestion
param managedIdentityClientId string
```
typo ^

File: infra/app/app-env.bicep [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+resource daprComponentPubsub 'Microsoft.App/managedEnvironments/daprComponents@2022-06-01-preview' = {
+  parent: caEnvironment
+  name: 'orderpubsub'
+  properties: {
+    componentType: 'pubsub.azure.servicebus'
+    version: 'v1'
+    metadata: [
+      {
+        name: 'azureClientId'
+        value: manegeIdentityClientId  // See https://docs.dapr.io/developing-applications/integrations/azure/authenticating-azure/#credentials-metadata-fields for MSI
```
Comment (paulyuk) [inline infra/app/app-env.bicep]:
```suggestion
        value: managedIdentityClientId  // See https://docs.dapr.io/developing-applications/integrations/azure/authenticating-azure/#credentials-metadata-fields for MSI
```

Comment (paulyuk) [inline infra/app/app-env.bicep]:
```suggestion
        value: managedIdentityClientId  // See https://docs.dapr.io/developing-applications/integrations/azure/authenticating-azure/#credentials-metadata-fields for MSI
```

File: infra/app/checkout.bicep [hunk 4]
```diff
@@ -0,0 +1,32 @@
+param location string = resourceGroup().location
```
Comment (paulyuk) [inline infra/app/checkout.bicep]: file should be named `publisher.bicep`

File: infra/app/checkout.bicep [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+param tags object = {}
+
+param containerAppsEnvironmentName string
+param containerRegistryName string
+param imageName string = ''
+param name string = ''
+param serviceName string = 'checkout'
```
Comment (paulyuk) [inline infra/app/checkout.bicep]: This is where the "domain object" of `checkout` rightly is set, versus name of file should should be more reusable

File: infra/app/checkout.bicep [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+param managedIdentityName string = ''
+
+module checkout '../core/host/container-app.bicep' = {
```
Comment (paulyuk) [inline infra/app/checkout.bicep]:
```suggestion
module publisher '../core/host/container-app.bicep' = {
```

File: infra/app/checkout.bicep [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    daprEnabled: true
+    containerName: serviceName
+    ingressEnabled: false
+    managedIdentityEnabled: true
+    managedIdentityName: managedIdentityName
+  }
+}
+
+
+output SERVICE_CHECKOUT_IMAGE_NAME string = checkout.outputs.imageName
```
Comment (paulyuk) [inline infra/app/checkout.bicep]:
```suggestion
output SERVICE_PUBLISHER_IMAGE_NAME string = checkout.outputs.imageName
```

File: infra/app/checkout.bicep [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+output SERVICE_CHECKOUT_NAME string = checkout.outputs.name
```
Comment (paulyuk) [inline infra/app/checkout.bicep]:
```suggestion
output SERVICE_PUBLISHER_NAME string = checkout.outputs.name
```

File: infra/app/checkout.parameters.json [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "name": {
+      "value": "${SERVICE_CHECKOUT_NAME}"
```
Comment (paulyuk) [inline infra/app/checkout.parameters.json]:
```suggestion
      "value": "${SERVICE_PUBLISHER_NAME}"
```

Whole file needs this change.  and upstream

File: infra/app/orders.bicep [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+param tags object = {}
+
+param containerAppsEnvironmentName string
+param containerRegistryName string
+param imageName string = ''
+param name string = ''
+param serviceName string = 'orders'
+param managedIdentityName string = ''
+
+module orders '../core/host/container-app.bicep' = {
```
Comment (paulyuk) [inline infra/app/orders.bicep]:
```suggestion
module subscriber '../core/host/container-app.bicep' = {
```

Whole file should rename references of `orders` to `subscriber`

File: infra/app/orders.bicep [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    containerName: serviceName
+    targetPort: 5001
+    managedIdentityEnabled: true
+    managedIdentityName: managedIdentityName
+  }
+}
+
+
+output ORDERS_URI string = orders.outputs.uri
+output SERVICE_ORDERS_IMAGE_NAME string = orders.outputs.imageName
```
Comment (paulyuk) [inline infra/app/orders.bicep]:
```suggestion
output SERVICE_SUBSCRIBER_IMAGE_NAME string = orders.outputs.imageName
```

File: infra/app/orders.parameters.json [hunk 12]
```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "name": {
+      "value": "${SERVICE_ORDERS_NAME}"
```
Comment (paulyuk) [inline infra/app/orders.parameters.json]:
```suggestion
      "value": "${SERVICE_SUBSCRIBER_NAME}"
```
Applies to whole file.. and upstream

Comment (paulyuk) [inline infra/app/orders.parameters.json]:
```suggestion
      "value": "${SERVICE_SUBSCRIBER_NAME}"
```
Applies to whole file.. and upstream

File: infra/app/servicebus-access.bicep [hunk 13]
```diff
@@ -0,0 +1,52 @@
+param managedIdentityName string
```
Comment (paulyuk) [inline infra/app/servicebus-access.bicep]: great!  just believe this belongs in /core/eventing/servicebus.  

And then the servicebus.bicep in /app probably should just do the access with this file?  or do it in app-env.bicep?

File: infra/core/database/cosmos/cosmos-account.bicep [hunk 14]
```diff
@@ -0,0 +1,48 @@
+param name string
```
Comment (paulyuk) [inline infra/core/database/cosmos/cosmos-account.bicep:1]: Why do we include /core/database/* in the PubSub sample?  I suggest we leave it out, so less to maintain.  this is dead code.

File: infra/core/host/appservice.bicep [hunk 15]
```diff
@@ -0,0 +1,97 @@
+param name string
```
Comment (paulyuk) [inline infra/core/host/appservice.bicep:1]: why include /core/host/appservice* ?

File: infra/core/host/staticwebapp.bicep [hunk 16]
```diff
@@ -0,0 +1,21 @@
+param name string
```
Comment (paulyuk) [inline infra/core/host/staticwebapp.bicep:1]: why include this file?

File: infra/core/host/functions.bicep [hunk 17]
```diff
@@ -0,0 +1,82 @@
+param name string
```
Comment (paulyuk) [inline infra/core/host/functions.bicep:1]: why include this file?

File: infra/core/storage/storage-account.bicep [hunk 18]
```diff
@@ -0,0 +1,38 @@
+param name string
```
Comment (paulyuk) [inline infra/core/storage/storage-account.bicep:1]: do we need this file?

File: infra/main.bicep [hunk 19]
```diff
@@ -2,51 +2,138 @@ targetScope = 'subscription'
...
+output SERVICE_ORDERS_NAME string = ordersApp.outputs.SERVICE_ORDERS_NAME
+output ORDERS_APP_URI string = ordersApp.outputs.ORDERS_URI
+output SERVICEBUS_ENDPOINT string = serviceBusResources.outputs.SERVICEBUS_ENDPOINT
+output SERVICEBUS_NAME string = serviceBusResources.outputs.serviceBusName
+output APPINSIGHTS_INSTRUMENTATIONKEY string = monitoring.outputs.applicationInsightsInstrumentationKey
+output APPLICATIONINSIGHTS_NAME string = monitoring.outputs.applicationInsightsName
+output AZURE_CONTAINER_ENVIRONMENT_NAME string = appEnv.outputs.environmentName
+output AZURE_CONTAINER_REGISTRY_ENDPOINT string = appEnv.outputs.registryLoginServer
+output AZURE_CONTAINER_REGISTRY_NAME string = appEnv.outputs.registryName
+output AZURE_MANAGED_IDENTITY_NAME string = serviceBusAccess.outputs.managedIdentityName
```
Comment (paulyuk) [inline infra/main.bicep:144]: Should we create user managed identity first in main, and then pass that to all the resources that need it?  it's weird to me that it originates from Service Bus (if I understand this).

**Review by paulyuk — CHANGES_REQUESTED** [review]
Awesome set of changes, thank you.  Just have some suggestions to make this more standard and reusable.  Also I err on the side of do not include bicep files we don't use, so we don't need to maintain.

File: infra/app/servicebus-access.bicep (hunk 13 above)
Comment (mukundansundar) [reply infra/app/servicebus-access.bicep]: For moving it into core eventing, will create a PR in azd repo first and then update it here later.
For now azd does not have it ... Would prefer to keep `core` folder as much as possible in sync with azd with minimal changes if any.

File: infra/core/database/cosmos/cosmos-account.bicep (hunk 14 above)
Comment (mukundansundar) [reply infra/core/database/cosmos/cosmos-account.bicep:1]: just to keep the `core` folder in sync with AZD repo one.

File: infra/core/host/appservice.bicep (hunk 15 above)
Comment (mukundansundar) [reply infra/core/host/appservice.bicep:1]: same as above

File: infra/main.bicep (hunk 19 above)
Comment (mukundansundar) [reply infra/main.bicep:139]: agreed changing it.

Comment (mukundansundar) [reply infra/main.bicep:144]: Have renamed `servicebus-access.bicep` to `access.bicep` as of now any type of access requests in other samples have been done as part of a module file only and not in `main.yaml`.

**Comment by mukundansundar** [comment]
> There have been a good amount of changes to the `containerapp` [bicep core modules](https://github.com/Azure/azure-dev/tree/main/templates/common/infra/bicep/core) from Azd over the last release (released today). I'd recommend you take a look at those and consider folding them back in here. A [migration guide](https://github.com/Azure/azure-dev/pull/2027) is available in the PR notes.
> 
> There is also a pending PR ([Azure/azure-dev#2157](https://github.com/Azure/azure-dev/pull/2157)) that enables dapr configuration as well.

Done.

**Comment by paulyuk** [comment]
@mukundansundar when I run `checkout` I see a new error.  Let me dig a little more.

✅  You're up and running! Both Dapr and your app logs will appear here.

== APP == INFO:root:Published data: {"orderId": 1}
❌  The App process exited with error code: signal: abort trap

**Review by paulyuk — APPROVED** [review]
Just did a deeper review, and tested in real azure subscriptions.  This is awesome!  thank you!

**Pull request merged**